mne/utils/numerics.py

```diff
@@ -498,15 +498,40 @@ def _time_mask(times, tmin=None, tmax=None, sfreq=None, raise_error=True):
     if raise_error and tmin > tmax:
         raise ValueError('tmin (%s) must be less than or equal to tmax (%s)'
                          % (orig_tmin, orig_tmax))
-    mask = (times >= tmin)
-    mask &= (times <= tmax)
+    mask = (times >= tmin) & (times <= tmax)
     if raise_error and not mask.any():
         raise ValueError('No samples remain when using tmin=%s and tmax=%s '
                          '(original time bounds are [%s, %s])'
                          % (orig_tmin, orig_tmax, times[0], times[-1]))
     return mask
 
 
+def _freq_mask(freqs, fmin=None, fmax=None, sfreq=None, raise_error=True):
+    """Safely find frequency boundaries."""
+    orig_fmin = fmin
+    orig_fmax = fmax
+    fmin = -np.inf if fmin is None else fmin
+    fmax = np.inf if fmax is None else fmax
+    if not np.isfinite(fmin):
+        fmin = freqs[0]
+    if not np.isfinite(fmax):
+        fmax = freqs[-1]
+    if sfreq is not None:
+        # Push 0.5/sfreq past the nearest frequency boundary first
```

Comment:
why do we need this?

Reply:
The ```# Push 0.5/sfreq past the nearest frequency boundary first``` part? As said to @agramfort, like in ```_time_mask``` it's needed if, for example, you want to include ```fmax=50```, but the corresponding frequency in freqs is something like 50.0001.

Reply:
I see this as unnecessarily complicated, why not just `np.argmin(np.abs(freq - fmin))`?

Reply:
There is probably some edge case that justified it for `_time_mask`. For now I'm fine having `_freq_mask` do the same thing as `_time_mask`, and if the simpler thing you suggest can be equivalent, let's change it in both places in another PR.

Reply:
good plan, I agree :)